internal/sql/fingerprint_test.go

```diff
@@ -0,0 +1,408 @@
+package sql
+
+import (
+	"testing"
+
+	"github.com/xwb1989/sqlparser"
+	"gotest.tools/v3/assert"
+)
+
+func TestExpressionString(t *testing.T) {
+	tests := []struct {
+		name     string
+		expr     expression
+		expected string
+	}{
+		{
+			name:     "Single AND term",
+			expr:     expression{{"_key LIKE 'match:1:*'", "_value > 10"}},
+			expected: "_key LIKE 'match:1:*' AND _value > 10",
+		},
+		{
+			name:     "Multiple AND terms in a single OR term",
+			expr:     expression{{"_key LIKE 'match:1:*'", "_value > 10", "_value < 5"}},
+			expected: "_key LIKE 'match:1:*' AND _value < 5 AND _value > 10",
+		},
+		{
+			name:     "Multiple OR terms with single AND terms",
+			expr:     expression{{"_key LIKE 'match:1:*'"}, {"_value < 5"}, {"_value > 10"}},
+			expected: "_key LIKE 'match:1:*' OR _value < 5 OR _value > 10",
+		},
+		{
+			name:     "Multiple OR terms with AND combinations",
+			expr:     expression{{"_key LIKE 'match:1:*'", "_value > 10"}, {"_value < 5", "_value > 0"}},
+			expected: "_key LIKE 'match:1:*' AND _value > 10 OR _value < 5 AND _value > 0",
+		},
+		{
+			name:     "Unordered terms",
+			expr:     expression{{"_value > 10", "_key LIKE 'match:1:*'"}, {"_value > 0", "_value < 5"}},
+			expected: "_key LIKE 'match:1:*' AND _value > 10 OR _value < 5 AND _value > 0",
+		},
+		{
+			name:     "Nested AND and OR terms with duplicates",
+			expr:     expression{{"_key LIKE 'match:1:*'", "_value < 5"}, {"_key LIKE 'match:1:*'", "_value < 5", "_value > 10"}},
+			expected: "_key LIKE 'match:1:*' AND _value < 5 OR _key LIKE 'match:1:*' AND _value < 5 AND _value > 10",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			assert.Equal(t, test.expected, test.expr.String())
+		})
+	}
+}
+
+func TestCombineOr(t *testing.T) {
+	tests := []struct {
+		name     string
+		a        expression
+		b        expression
+		expected expression
+	}{
+		{
+			name:     "Combining two empty expressions",
+			a:        expression([][]string{}),
+			b:        expression([][]string{}),
+			expected: expression([][]string{}),
+		},
+		// {
+		// 	name:     "Annulment law",
+		// 	a:        Expression([][]string{{"_value > 10"}}),
+		// 	b:        Expression([][]string{{}}), // equivalent to 1
+		// 	expected: Expression([][]string{{}}),
+		// },
```

Comment:
Why is this code commented? Do we need this?

Reply:
since solution was moving towards boolean algebra, I was trying to accomodate the laws. Its true some of them doesn't make sense for our use case like `Identity law`, `Annulment law` etc. 
I'll remove this commented test case. There are few more test cases commented below, I feel they are important and should be supported in future thus not removing them.

Reply:
Thanks for explaining.